File: c/sorting/Selection_Sort.c
#include <stdio.h>

void swap(int *x, int *y){
    int temp = *x; 
    *x = *y, *y = temp;
}
 
void selectionSort(int* arr, int n){
    int i, j, min_idx;
    for (i = 0; i < n - 1; i++){
        min_idx = i;
        for (j = i + 1; j < n; j++)
          if (arr[j] < arr[min_idx])
            min_idx = j;
        swap(&arr[min_idx], &arr[i]);
    }
}

int main(){

    return 0;
}

File: java/sorting/Selection_Sort.java
package Algorithms_Code.Java;

public class Selection_Sort {   
    
    void selection_sort(int arr[]){
        int n = arr.length;
        for (int i = 0; i < n - 1; i++){
            int min_idx = i;
            for (int j = i + 1; j < n; j++)
                if (arr[j] < arr[min_idx])
                    min_idx = j;
            int temp = arr[min_idx];
            arr[min_idx] = arr[i];
            arr[i] = temp;
        }
    }
        
    public static void main(String args[]){
        
    }

}
